microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/reactivemessaging/java/MicroProfileReactiveMessagingASTValidator.java

```diff
@@ -0,0 +1,85 @@
+* http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+* which is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lsp4mp.jdt.internal.reactivemessaging.java;
+
+import static org.eclipse.lsp4mp.jdt.internal.reactivemessaging.MicroProfileReactiveMessagingConstants.MICRO_PROFILE_REACTIVE_MESSAGING_DIAGNOSTIC_SOURCE;
+import static org.eclipse.lsp4mp.jdt.internal.reactivemessaging.MicroProfileReactiveMessagingConstants.INCOMING_ANNOTATION;
+import static org.eclipse.lsp4mp.jdt.internal.reactivemessaging.MicroProfileReactiveMessagingConstants.OUTGOING_ANNOTATION;
+
+import java.util.List;
+import java.util.logging.Logger;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.dom.SingleMemberAnnotation;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.DiagnosticSeverity;
+import org.eclipse.lsp4mp.jdt.core.java.diagnostics.JavaDiagnosticsContext;
+import org.eclipse.lsp4mp.jdt.core.java.validators.JavaASTValidator;
+import org.eclipse.lsp4mp.jdt.core.utils.JDTTypeUtils;
+
+/**
+ *
+ * MicroProfile Reactive Messaging Diagnostics:
+ *
+ * <ul>
+ * <li>Diagnostic: display reactive messaging diagnostic message if the name of
+ * the consumed channel is blank.</li>
+ * </ul>
+ *
+ * <p>
+ * This rule comes from
+ * https://github.com/eclipse/microprofile-reactive-messaging/blob/62c9ed5dffe01125941bb185f1433d6307b83c86/api/src/main/java/org/eclipse/microprofile/reactive/messaging/Incoming.java#L95
+ * </p>
+ * 
+ * @See https://github.com/eclipse/microprofile-reactive-messaging
+ *
+ */
+public class MicroProfileReactiveMessagingASTValidator extends JavaASTValidator {
+
+    private static final String BLANK_CHANNEL_NAME_MESSAGE = "The name of the consumed channel must not be blank.";
+
+    private static final Logger LOGGER = Logger.getLogger(MicroProfileReactiveMessagingASTValidator.class.getName());
+
+    @Override
+    public void initialize(JavaDiagnosticsContext context, List<Diagnostic> diagnostics) {
+        super.initialize(context, diagnostics);
+    }
+
+    @Override
+    public boolean isAdaptedForDiagnostics(JavaDiagnosticsContext context,
+            IProgressMonitor monitor) throws CoreException {
+        IJavaProject javaProject = context.getJavaProject();
+        boolean adapted = JDTTypeUtils.findType(javaProject, INCOMING_ANNOTATION) != null
+                || JDTTypeUtils.findType(javaProject, OUTGOING_ANNOTATION) != null;
+        return adapted;
+    }
+
+    @Override
+    public boolean visit(SingleMemberAnnotation node) {
+        validateIncomingOutgoingAnnotation(node);
+        super.visit(node);
+        return true;

```

Comment:
I don't think you need these. In fact your could return `false` here. Although I notice a few example in the code that do exactly the same thing.

`super.visite(node)` just calls the base class, which if you look at, does nothing other than return a value, which we're ignoring here.

The boolean return value from a visit call indicates whether the caller should continue to visit the children of this element. This might be useful if some of the elements you care about are children of `SingleMemberAnnotation`. In our case, we know we won't find anything there, so we may as well return `false` and not traverse any deeper. Usually the reason to do this is performance.